pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -281,8 +275,11 @@ func (s *Service) sendActionEvent(msg dispatcher.DIDCommMsg, aEvent chan<- dispa
 	// TODO pass invitation id #397
 	didCommAction := dispatcher.DIDCommAction{
 		Message: msg,
-		Callback: func(didCommCallback dispatcher.DIDCommCallback) {
-			s.processCallback(id, didCommCallback)
+		Continue: func() {
+			s.processCallback(id)
+		},
+		Fail: func(err error) {
```

Comment:
This `Fail` function should also be marking the state machine instance as 'stale' somehow

Reply:
Added [TODO](https://github.com/hyperledger/aries-framework-go/pull/437/files#diff-f761c91e753eba1c42a68076df06ce12R333). Also we would need to clean-up temp data.

Reply:
Is this the same TODO @rolsonquadras @llorllale https://github.com/hyperledger/aries-framework-go/pull/437#discussion_r332190493

Reply:
@troyronda yes.